### PR TITLE
refactor signatures

### LINKED ISSUE #1404: Signature refactor
In our backends, we are using the following logic to sign requests:

```
web3.eth.sign(encode_defunct(message), wallet)  (py)
web3.eth.sign(message, wallet)   (js)
```

which is incompatible with how signatures are checked in solidity

Proposal:
 - generate the hash first, by converting message to bytes and using keccak
- sign the hash

ie:
```
consumerMessage = web3.utils.soliditySha3({
      t: 'bytes',
      v: web3.utils.utf8ToHex(message)
 })
consumerSignature = web3.eth.sign(consumerMessage, wallet)
```

This requires a refactor on multiple components, and thus they need to be merged/deployed in the same time.

Affected repos:

- aquarius
- provider
- ocean.js
- ocean.py
- operator-service
- pod-configuration

(see EPIC https://github.com/oceanprotocol/multi-repo-issue/issues/144)

### PR DESCRIPTION
**Pull request by alexcos20**
Closes #1404

### CODE REVIEW

**Comment by alexcos20** [comment]
@kremalicious - any reason for putting it back to draft?

**Comment by kremalicious** [comment]
the `DoNotMerge` label combined with lots of commented out code, and making CI using custom branches for external components made it look like a draft but get now that some of those things are because external components need update too

File: src/provider/Provider.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {
   ProviderInitialize
 } from '../@types/'
 import { noZeroX } from '../utils/ConversionTypeHelper'
-import { signText, signWithHash } from '../utils/SignatureUtils'
+// import { signText, signWithHash } from '../utils/SignatureUtils'
```
Comment (mihaisc) [inline src/provider/Provider.ts:13]: don't comment, just remove if not needed.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -119,6 +119,9 @@ jobs:
           bash -x start_ocean.sh --with-provider2 --no-dashboard --with-c2d 2>&1 > start_ocean.log &
         env:
           CONTRACTS_VERSION: v1.0.0-alpha.28
+          PROVIDER_VERSION: refactor_signatures
+          AQUARIUS_VERSION: refactor_signatures
+          OPERATOR_SERVICE_VERSION: refactor_signatures
```
Comment (mihaisc) [inline .github/workflows/ci.yml:124]: we shouldn't have this branches here.

File: src/utils/SignatureUtils.ts [hunk 3]
```diff
@@ -26,6 +27,7 @@ export async function signText(
     }
   }
 }
+*/
 
 export async function signHash(web3: Web3, message: string, address: string) {
```
[the remark this answers is not in the document]
Comment (alexcos20) [reply src/utils/SignatureUtils.ts:4]: signHash is used in:
  - test/unit/NftFactory.test.ts
  - test/unit/Datatoken.test.ts

File: .github/workflows/ci.yml (hunk 2 above)
Comment (alexcos20) [reply .github/workflows/ci.yml:124]: Will change that as soon as we merge them and release new versions.
If I will change it now, all tests are going to fail.
We need to merge all PRs for all components in the same time

File: src/provider/Provider.ts [hunk 4]
```diff
@@ -490,12 +487,17 @@ export class Provider {
     signatureMessage += jobId || ''
     signatureMessage += (did && `${noZeroX(did)}`) || ''
     signatureMessage += nonce
-    const signature = await this.createHashSignature(
+    /* const signature = await this.createHashSignature(
```
Comment (bogdanfazakas) [inline src/provider/Provider.ts:490]: Think you missed this comments 😃

File: src/provider/Provider.ts [hunk 5]
```diff
@@ -620,7 +622,8 @@ export class Provider {
     signatureMessage += jobId
     signatureMessage += index.toString()
     signatureMessage += nonce
-    const signature = await this.createHashSignature(web3, accountId, signatureMessage)
+    // const signature = await this.createHashSignature(web3, accountId, signatureMessage)
```
Comment (bogdanfazakas) [inline src/provider/Provider.ts:625]: same as above

File: src/provider/Provider.ts [hunk 6]
```diff
@@ -680,12 +683,17 @@ export class Provider {
     signatureMessage += jobId || ''
     signatureMessage += (did && `${noZeroX(did)}`) || ''
     signatureMessage += nonce
-    const signature = await this.createHashSignature(
+    /* const signature = await this.createHashSignature(
```
Comment (bogdanfazakas) [inline src/provider/Provider.ts:686]: this one also 😃

File: src/provider/Provider.ts (hunk 4 above)
Comment (alexcos20) [reply src/provider/Provider.ts]: good catch :)

**Review by bogdanfazakas — APPROVED** [review]

**Pull request merged**